packages/cli/yaml/docs-validator/src/rules/valid-file-types/valid-file-types.ts

```diff
@@ -0,0 +1,77 @@
+import { doesPathExist } from "@fern-api/fs-utils";
+import { fileTypeFromBuffer } from "file-type";
+import { readFile } from "fs/promises";
+import isSvg from "is-svg";
+import { Rule } from "../../Rule";
+
+const ALLOWED_FILE_TYPES = [
+    // image files
+    "image/jpeg",
+    "image/png",
+    "image/gif",
+    "image/webp",
+    "image/tiff",
+    "image/x-icon",
+    // video files
+    "video/quicktime",
+    "video/mp4",
+    "video/webm",
+    // audio files
+    "audio/mpeg",
+    "audio/ogg",
+    "audio/wav",
+    // document files
+    "application/pdf",
+    // font files
+    "font/woff",
+    "font/woff2",
+    "font/otf",
+    "font/ttf"
+];
+
+export const ValidFileTypes: Rule = {
+    name: "valid-file-types",
+    create: () => {
+        return {
+            filepath: async ({ absoluteFilepath, value, willBeUploaded }) => {
+                if (!willBeUploaded) {
+                    return [];
+                }
+
+                const doesExist = await doesPathExist(absoluteFilepath);
+
+                if (doesExist) {
+                    const file = await readFile(absoluteFilepath);
+                    const isValid = await isValidFileType(file);
+
+                    if (!isValid) {
+                        return [
+                            {
+                                severity: "error",
+                                message: `File type of ${value} is invalid`
+                            }
+                        ];
+                    }
+                }
+
+                return [];
+            }
+        };
+    }
+};
+
+export const isValidFileType = async (file: Buffer): Promise<boolean> => {
```

Comment:
we're still missing javascript and css files

Reply:
Are we uploading JS and CSS files directly to S3?

Those files can't be checked the same way that we're checking binary files in this function, since they're just text. One solution would be to just say "if the file type checker can't derive a mime type from the file, let it pass". This would allow people to upload arbitrary PHP or other text-based files as well, but I can't think of a better solution at the moment.

Reply:
@trevorblades can we fallback to extension checking if it can't derive a mime type? not 100% fullproof but still better